tests/end_to_end/requirements.txt

```diff
@@ -26,4 +26,4 @@ rich==13.5.2
 six==1.16.0
 SQLAlchemy
 typeguard==2.13.3
-typing_extensions==4.7.1
+typing_extensions==4.10.0
```

Comment:
context from standup, the latest is `4.12.2` but not sure how that affects things yet. can add later. 
<img width="607" alt="image" src="https://github.com/user-attachments/assets/586c57f2-cd78-441e-a5de-09d58cd48539">